File: fitness-service/src/main/java/com/frex/fitness/service/BasicInfoService.java
package com.frex.fitness.service;

import com.frex.fitness.exception.BasicInfoAlreadySavedException;
import com.frex.fitness.exception.BasicInfoDoesNotExistException;
import com.frex.fitness.model.BasicInfo;

public interface BasicInfoService {
    //Create
    boolean saveBasicInfo(BasicInfo basicInfo) throws BasicInfoAlreadySavedException;
    //Retrive
    BasicInfo retrieveBasicInfo(String userId) throws BasicInfoDoesNotExistException;
    //Update
    boolean updateBasicInfo(BasicInfo basicInfo) throws BasicInfoDoesNotExistException;
    //Delete
    boolean deleteBasicInfo(String userId);
    //Check
    boolean checkIfExists(String userId);
}

File: frex-ui/src/app/services/fitness/basic-info.service.ts
import {Injectable} from '@angular/core';
import {environment} from '../../../environments/environment';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {BasicInfo} from '../../util/fitness-activities';

@Injectable({
  providedIn: 'root'
})
export class BasicInfoService {
  private url = environment.fitnessServiceURL + 'basicInfo/';

  constructor(
    private httpClient: HttpClient
  ) {
  }

  async saveBasicInfo(basicInfo: BasicInfo) {
    return await this.httpClient.post(this.url + 'save', basicInfo).toPromise().then(
      (result) => {
        return result;
      });
  }

  retrieveBasicInfo(userId: string) {
    return this.httpClient.get(this.url + 'retrieve', {
      headers: new HttpHeaders().set('userId', userId)
    });
  }

  updateBasicInfo(basicInfo: BasicInfo) {
    return this.httpClient.put(this.url + 'update', basicInfo);
  }

  async checkIfExists(userId: string) {
    return await this.httpClient.get(this.url + 'check', {
      headers: new HttpHeaders().set('userId', userId)
    }).toPromise().then(
      result => {
        return result;
      }
    );
  }
}
